pkg/build/util/generate.go

```diff
@@ -27,9 +27,14 @@ func GenerateBuildFromConfig(bc *buildapi.BuildConfig, r *buildapi.SourceRevisio
 			Revision: r,
 		},
 		ObjectMeta: kapi.ObjectMeta{
-			Labels: map[string]string{buildapi.BuildConfigLabel: bcCopy.Name},
+			Labels: bcCopy.Labels,
```

Comment:
you can see the whole scope of bcCopy here.  The copy is created in this function and not returned out of it.  The only reason the copy is being made is so we can assign fields from it into the build we are creating w/o them sharing data.